web/oxo-web-ben2/README.md

```diff
@@ -0,0 +1,403 @@
+# Security Vulnerabilities Report - SAML Todo Application
+
+## Critical Vulnerabilities
+
+### 1. SAML Assertion Replay Attack - CRITICAL
+
+**Description**:
+The application doesn't track processed SAML assertion IDs or validate timestamps, allowing attackers to reuse captured SAML responses.
+
+**Location**: `app.py:144-182`
+```python
+@app.route('/saml/acs', methods=['POST'])
+def saml_acs():
+    auth.process_response()
+    # No assertion ID tracking
+    # No NotOnOrAfter validation
+    # No NotBefore validation
+```
+
+**Risk**:
+Attackers can:
+- Capture valid SAML responses via man-in-the-middle attacks
+- Replay authentication tokens indefinitely
+- Bypass authentication without valid credentials
+- Gain unauthorized access to user accounts
+- Reuse single token across multiple sessions
```

Comment:
A more common and critical bug is when the app accepts mulitple SAML each for a different group and once saml from org "hacker" can be used to authenticate as user "target"